borsh-rs/borsh/src/de/mod.rs

```diff
@@ -31,11 +31,11 @@ pub trait BorshDeserialize: Sized {
         None
     }
 
-    /// Whether it's possible to transmute `Vec<u8>` directly into `Vec<Self>`.
-    /// If it's `true`, then `Vec<Self>` implementation will use unsafe behavior to transmute
-    /// `Vec<u8>` to `Vec<Self>`. Required `size_of::<Self>() == size_of::<u8>()`.
+    /// Whether Self is u8.
+    /// NOTE: `Vec<u8>` is the most common use-case for serialization and deserialization, it's
+    /// worth handling it as a special case to improve performance.
```

Comment:
Please explain that it is to workaround absence of impl specialization and link the Rust issue.

Reply:
Added